File: utilslibrary/src/main/java/com/sincerity/utilslibrary/http/IHttpEngine.java
package com.sincerity.utilslibrary.http;

import java.util.Map;

/**
 * Created by Sincerity on 2019/9/3.
 * 描述：引擎通用的请求方式
 */
public interface IHttpEngine {
    /**
     * post请求
     *
     * @param url
     * @param params
     * @param httpCallBack
     */
    void post(String url, Map<String, String> params, HttpCallBack httpCallBack);

    /**
     * get请求
     *
     * @param url
     * @param params
     * @param httpCallBack
     */
    void get(String url, Map<String, String> params, HttpCallBack httpCallBack);
}

File: utilslibrary/src/main/java/com/sincerity/utilslibrary/httputils/IHttpEngine.kt
package com.sincerity.utilslibrary.httputils

import android.content.Context
import org.jetbrains.annotations.NotNull

/**
 * Created by Sincerity on 2019/11/7.
 * 描述：网络引擎的规范
 */
 interface IHttpEngine {

    //get
    fun get(context: Context, cache:Boolean,@NotNull url: String, prams: Map<String, Any>, callBack: CallBack?)

    //post
    fun post(context: Context,cache:Boolean, @NotNull url: String, prams: Map<String, Any>, callBack: CallBack?)
    //下载文件
    //上传文件
}